server/lib/availabilitySync.ts

```diff
@@ -57,141 +54,93 @@ class AvailabilitySync {
+            media.status4k === MediaStatus.AVAILABLE ||
+            media.status4k === MediaStatus.PARTIALLY_AVAILABLE
+          ) {
+            (media.status4k = mediaStatus4k),
+              (media.serviceId4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.serviceId4k
+                  : null),
+              (media.externalServiceId4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId4k
+                  : null),
+              (media.externalServiceSlug4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug4k
+                  : null),
+              (media.ratingKey4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.ratingKey4k
+                  : null);
+          }
 
-          let didDeleteSeasons = false;
-          for (const season of seasons) {
-            if (
-              !mediaExists &&
-              (season.status !== MediaStatus.UNKNOWN ||
-                season.status4k !== MediaStatus.UNKNOWN)
-            ) {
-              await seasonRepository.update(
-                { id: season.id },
-                {
-                  status: MediaStatus.UNKNOWN,
-                  status4k: MediaStatus.UNKNOWN,
-                }
-              );
-            } else {
-              const seasonExists = await this.seasonExists(media, season);
-
-              if (!seasonExists) {
-                logger.info(
-                  `Removing season ${season.seasonNumber}, media ID ${media.id} because it does not exist in any of your media instances.`,
-                  { label: 'AvailabilitySync' }
-                );
-
-                if (
-                  season.status !== MediaStatus.UNKNOWN ||
-                  season.status4k !== MediaStatus.UNKNOWN
-                ) {
-                  await seasonRepository.update(
-                    { id: season.id },
-                    {
-                      status: MediaStatus.UNKNOWN,
-                      status4k: MediaStatus.UNKNOWN,
-                    }
-                  );
-                }
-
-                const seasonToBeDeleted = await seasonRequestRepository.findOne(
-                  {
-                    relations: {
-                      request: {
-                        media: true,
-                      },
-                    },
-                    where: {
-                      request: {
-                        media: {
-                          id: media.id,
-                        },
-                      },
-                      seasonNumber: season.seasonNumber,
-                    },
-                  }
-                );
-
-                if (seasonToBeDeleted) {
-                  await seasonRequestRepository.remove(seasonToBeDeleted);
-                }
-
-                didDeleteSeasons = true;
-              }
-            }
```

Comment:
All season handling has been moved to the existsInSonarr function. This simplifies the flow of checking through each show.